engine/execution/computation/computer/computer.go

```diff
@@ -46,14 +49,15 @@ type BlockComputer interface {
 }
 
 type blockComputer struct {
-	vm                    VirtualMachine
-	vmCtx                 fvm.Context
-	metrics               module.ExecutionMetrics
-	tracer                module.Tracer
-	log                   zerolog.Logger
-	systemChunkCtx        fvm.Context
-	committer             ViewCommitter
-	executionDataProvider *provider.Provider
+	vm                         VirtualMachine
+	vmCtx                      fvm.Context
+	metrics                    module.ExecutionMetrics
+	tracer                     module.Tracer
+	log                        zerolog.Logger
+	systemChunkCtx             fvm.Context
+	committer                  ViewCommitter
+	executionDataProvider      *provider.Provider
+	reusableCadenceRuntimePool fvm.ReusableCadenceRuntimePool
```

Comment:
Should the VM be responsible for holding this, since its already holding the runtime?